00-init.tf

```diff
@@ -4,4 +4,11 @@ terraform {
 
 provider "azurerm" {
   alias = "hmcts-control"
+  features {}
+}
+
+provider "azurerm" {
+  features {}
+  alias           = "loganalytics"
+  subscription_id = local.log_analytics_workspace[[for x in keys(local.log_analytics_env_mapping) : x if contains(local.log_analytics_env_mapping[x], var.environment)][0]].subscription_id
```

Comment:
a provider that is not solely an alias should always be passed in from the parent.

i.e. don't add this here, pass down the workspace ID from the calling code